Best practices/Serialization & Deserialization.md

```diff
@@ -0,0 +1,107 @@
+##  System.Text.Json
+
+Serialization and deserialization are fundamental processes in software development, allowing for the conversion of objects to a format that can be easily stored or transmitted and then reconstructed later. In the .NET ecosystem, while there have been several libraries available for these operations, `System.Text.Json` has emerged as a modern,  high-performance option built directly into .NET Core.
+
+For performance-critical scenarios  using `System.Text.Json` source generators  can provide significant improvements by generating optimized serialization code at compile-time.
+
+### Basic Usage
+
+To convert an object into its JSON representation:
+
+```string jsonString = JsonSerializer.Serialize(yourObject);```
+
+To convert a JSON string back into an object:
+
+```YourObjectType obj = JsonSerializer.Deserialize<YourObjectType>(jsonString);```
+	
+	
+### JsonSerializerOptions
+`JsonSerializerOptions` is a class that provides a way to specify various serialization and deserialization behaviors. Here are some of the commonly used options:
+
+**PropertyNameCaseInsensitive**: When set to `true`, property name matching during deserialization is case-insensitive. This can be useful when dealing with JSON from sources that don't maintain consistent casing.
+    
+```var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };```
+
+ **WriteIndented**: Determines whether the output JSON should be pretty-printed with indentation, which can be useful for debugging or human readability.
+
+```var options = new JsonSerializerOptions { WriteIndented = true };```
+    
+**DefaultIgnoreCondition**: Determines the condition under which a property will be ignored during serialization. For instance, you can set it to `JsonIgnoreCondition.WhenWritingNull` to ignore properties with `null` values.
+   
+```var options = new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull };```
+        
+**AllowTrailingCommas**: When set to `true`, the deserializer will tolerate trailing commas in the JSON.
+    
+```var options = new JsonSerializerOptions { AllowTrailingCommas = true };``` 
+    
+**MaxDepth**: Sets the maximum depth allowed when reading or writing JSON, protecting against deeply nested structures.
+    
+```var options = new JsonSerializerOptions { MaxDepth = 10 };```
+    
```

Comment:
Maybe we can elaborate this one, what 10 here mean? Can we describe use case when we want 1 to set and when we may need 10 :D